opennms-config/src/main/java/org/opennms/netmgt/config/KSC_PerformanceReportFactory.java

```diff
@@ -397,7 +397,7 @@ public static synchronized void getBeginEndTime(String interval, Calendar begin_
                 end_time.add(Calendar.DATE, -1);
                 end_time.set(Calendar.HOUR_OF_DAY, 22);
             } else if (interval.equals("This Week") || interval.equals("Last Week")) {
-                begin_time.set(Calendar.DAY_OF_WEEK, 1);
+                begin_time.set(Calendar.DAY_OF_WEEK, begin_time.getFirstDayOfWeek());
                 end_time.set(Calendar.DAY_OF_WEEK, 7);
```

Comment:
I just noticed that this needs to change as well. You need to calculate the end of the week day based on the beginning of the week day, otherwise you'll only have 6 days in your week.